src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -31,6 +31,12 @@ public class RestoreCommand
         private readonly Dictionary<RestoreTargetGraph, Dictionary<string, LibraryIncludeFlags>> _includeFlagGraphs
             = new Dictionary<RestoreTargetGraph, Dictionary<string, LibraryIncludeFlags>>();
 
+        public KeyValuePair<string, Guid> ParentId { get; }
+
+        public KeyValuePair<string, Guid> RestoreId = new KeyValuePair<string, Guid>(nameof(RestoreId), Guid.NewGuid());
```

Comment:
why not being consistent with OperationId

Reply:
beacuse RestoreId is the child of OperationId, if everyone have same name, then you need to join every table to find the relation